### PR TITLE
implemented and designed the task details page

### PR DESCRIPTION
**Pull request by farhanp1502**
Added the feature of task details with sub-task.

### CODE REVIEW

File: projects/project-player-library/src/lib/pages/details-page/details-page.component.ts [hunk 1]
```diff
@@ -2,7 +2,7 @@ import { Component, OnInit } from '@angular/core';
 import { MatDialog } from '@angular/material/dialog';
 import { Router } from '@angular/router';
 import { DailogPopupComponent } from '../../shared/dialog-popup/dailog-popup.component';
-import { projectDetailsData } from './project-details.component.spec.data';
+import { projectDetailsData } from '../../project-details.component.spec.data';
```
Comment (Vinod-V3) [inline projects/project-player-library/src/lib/pages/details-page/details-page.component.ts:5]: Why is the path changed?

File: projects/project-player-library/src/lib/pages/details-page/details-page.component.ts
Comment (Vinod-V3) [inline projects/project-player-library/src/lib/pages/details-page/details-page.component.ts:1]: Majority of the changes are formatting changes like adding spaces, changing from double quotes to single quotes. These are unnecessary changes, please revert it back

File: projects/project-player-library/src/lib/pages/task-details-page/task-details-page.component.html [hunk 2]
```diff
@@ -1 +1,50 @@
...
+        <h3>{{"SUBTASKS"|translate}}</h3>
+        <div class="subtask-input">
+            <mat-icon fontIcon="subdirectory_arrow_right"></mat-icon>
+            <div class="subtask-input-field">
+
+                    <mat-form-field class="example-full-width">
+                        <input #subTask type="text" matInput [formControl]="textFormControl"
+                                [placeholder]="'ADD_SUBTASK_NAME'|translate">
+                    </mat-form-field>
+                    <button [disabled]="subTask.value === ''" mat-flat-button color="primary" (click)="addSubTask(subTask.value)" >{{"ADD_SUBTASKS"|translate}}</button>
```
Comment (Vinod-V3) [inline projects/project-player-library/src/lib/pages/task-details-page/task-details-page.component.html:34]: change the condition from subTask.value === '' to !subTask.value and check

File: projects/project-player-library/src/lib/pages/task-details-page/task-details-page.component.ts [hunk 3]
```diff
@@ -1,20 +1,114 @@
...
 })
-export class TaskdetailspageComponent implements OnInit{
-  constructor(private route:ActivatedRoute){}
+export class TaskDetailsPageComponent implements OnInit {
+  constructor(private route: ActivatedRoute, private dialog: MatDialog) {}
+  taskId: any;
+  selectedValue!: string;
+  textFormControl = new FormControl('');
+  task: any = {};
+  taskOptions = [
```
Comment (Vinod-V3) [inline projects/project-player-library/src/lib/pages/task-details-page/task-details-page.component.ts:20]: Change the key from viewValue to label

File: projects/project-player-library/src/lib/pages/task-details-page/task-details-page.component.ts [hunk 4]
```diff
@@ -1,20 +1,114 @@
...
+    },
+    {
+      viewValue: 'Completed',
+      value: 'completed',
+    },
+  ];
+
   ngOnInit(): void {
-    this.route.paramMap.subscribe((params:any)=>{
+    this.route.paramMap.subscribe((params: any) => {
```
Comment (Vinod-V3) [inline projects/project-player-library/src/lib/pages/task-details-page/task-details-page.component.ts:36]: Revert this formatting change

File: projects/project-player-library/src/lib/pages/task-details-page/task-details-page.component.ts [hunk 5]
```diff
@@ -1,20 +1,114 @@
...
+export class TaskDetailsPageComponent implements OnInit {
+  constructor(private route: ActivatedRoute, private dialog: MatDialog) {}
+  taskId: any;
+  selectedValue!: string;
+  textFormControl = new FormControl('');
+  task: any = {};
+  taskOptions = [
+    {
+      viewValue: 'Not Started',
+      value: 'NOTstarted',
```
Comment (Vinod-V3) [inline projects/project-player-library/src/lib/pages/task-details-page/task-details-page.component.ts:23]: why the value is given as NOTstarted instead of notStarted

File: projects/project-player-library/src/lib/pages/task-details-page/task-details-page.component.ts [hunk 6]
```diff
@@ -1,20 +1,114 @@
...
+    this.textFormControl.reset();
+  }
+  editTask() {
+    this.openEditTaskName('0', '0', this.task.name,"EDIT_TASK"
+    );
+  }
+  openEditTaskName(
+    enterAnimationDuration: string,
+    exitAnimationDuration: string,
+    taskName: string,
```
Comment (Vinod-V3) [inline projects/project-player-library/src/lib/pages/task-details-page/task-details-page.component.ts:61]: Giving taskName and editType here, is it required??

File: projects/project-player-library/src/lib/project-details.component.spec.data.ts
Comment (Vinod-V3) [inline projects/project-player-library/src/lib/project-details.component.spec.data.ts:1]: Any changes made on this spec.data file need not be comitted from now, its only for local testing purpose
We will do only when its required

File: projects/project-player-library/src/lib/shared/edit-task-card/edit-task-card.component.ts [hunk 7]
```diff
@@ -0,0 +1,28 @@
...
+
+@Component({
+  selector: 'lib-edit-task-card',
+  templateUrl: './edit-task-card.component.html',
+  styleUrl: './edit-task-card.component.css'
+})
+export class EditTaskCardComponent {
+  title:string='';
+  editType:string='';
+  @Output() editname:EventEmitter<string>= new EventEmitter<string>();
```
Comment (Vinod-V3) [inline projects/project-player-library/src/lib/shared/edit-task-card/edit-task-card.component.ts:13]: Use camel case naming convention

File: projects/project-player-library/src/lib/shared/subtask-card/subtask-card.component.ts [hunk 8]
```diff
@@ -0,0 +1,88 @@
...
+  styleUrl: './subtask-card.component.css'
+})
+export class SubtaskCardComponent {
+@Input() subTask: any;
+@Output() deleteSubTaskEvent = new EventEmitter<string>();
+@Output()  updateSubTaskStatusEvent = new EventEmitter<any>();
+
+subTaskOptions = [
+  {
+    viewValue : "Not Started",
```
Comment (Vinod-V3) [inline projects/project-player-library/src/lib/shared/subtask-card/subtask-card.component.ts:18]: Change the key from viewValue to label

File: projects/project-player-library/src/lib/shared/subtask-card/subtask-card.component.ts [hunk 9]
```diff
@@ -0,0 +1,88 @@
...
+    width: '400px',
+    enterAnimationDuration,
+    exitAnimationDuration,
+  });
+  modelref.componentInstance.title = subTaskName;
+  modelref.componentInstance.editType=editType;
+  modelref.componentInstance.editname.subscribe((res)=>{
+    this.subTask.name = res;
+  })
+  modelref.afterClosed().subscribe((res: boolean) => {
```
Comment (Vinod-V3) [inline projects/project-player-library/src/lib/shared/subtask-card/subtask-card.component.ts:77]: What actions will you be handling within the if else condition here??

File: projects/project-player-library/src/lib/pages/details-page/details-page.component.ts (hunk 1 above)
Comment (farhanp1502) [reply projects/project-player-library/src/lib/pages/details-page/details-page.component.ts:5]: it was coming from the details page which was in the page folder and it is overall data so i made it from parent folder

File: projects/project-player-library/src/lib/pages/details-page/details-page.component.ts
Comment (farhanp1502) [reply projects/project-player-library/src/lib/pages/details-page/details-page.component.ts:1]: sure

File: projects/project-player-library/src/lib/pages/main-player/main-player.component.css
[the remark this answers is not in the document]
Comment (farhanp1502) [reply projects/project-player-library/src/lib/pages/main-player/main-player.component.css:1]: done

File: projects/project-player-library/src/lib/pages/task-details-page/task-details-page.component.html (hunk 2 above)
Comment (farhanp1502) [reply projects/project-player-library/src/lib/pages/task-details-page/task-details-page.component.html:34]: done

File: projects/project-player-library/src/lib/pages/task-details-page/task-details-page.component.ts (hunk 3 above)
Comment (farhanp1502) [reply projects/project-player-library/src/lib/pages/task-details-page/task-details-page.component.ts:20]: done

File: projects/project-player-library/src/lib/pages/task-details-page/task-details-page.component.ts (hunk 5 above)
Comment (farhanp1502) [reply projects/project-player-library/src/lib/pages/task-details-page/task-details-page.component.ts:23]: done

File: projects/project-player-library/src/lib/pages/task-details-page/task-details-page.component.ts (hunk 6 above)
Comment (farhanp1502) [reply projects/project-player-library/src/lib/pages/task-details-page/task-details-page.component.ts:61]: 1. taskName this is for editing of the task name or subtask name 
2. editType is type of modal used for task or subtask

File: projects/project-player-library/src/lib/project-details.component.spec.data.ts
Comment (farhanp1502) [reply projects/project-player-library/src/lib/project-details.component.spec.data.ts:1]: okay

File: projects/project-player-library/src/lib/shared/edit-task-card/edit-task-card.component.ts (hunk 7 above)
Comment (farhanp1502) [reply projects/project-player-library/src/lib/shared/edit-task-card/edit-task-card.component.ts:13]: done

File: projects/project-player-library/src/lib/shared/subtask-card/subtask-card.component.ts (hunk 8 above)
Comment (farhanp1502) [reply projects/project-player-library/src/lib/shared/subtask-card/subtask-card.component.ts:18]: done

File: projects/project-player-library/src/lib/shared/subtask-card/subtask-card.component.ts (hunk 9 above)
Comment (farhanp1502) [reply projects/project-player-library/src/lib/shared/subtask-card/subtask-card.component.ts:77]: Here the subtask data will be updated for the indexedDB .

File: projects/project-player-library/src/lib/pages/task-details-page/task-details-page.component.ts (hunk 4 above)
Comment (farhanp1502) [reply projects/project-player-library/src/lib/pages/task-details-page/task-details-page.component.ts:36]: done

File: projects/project-player-library/src/lib/pages/details-page/details-page.component.ts [hunk 10]
```diff
@@ -1,9 +1,11 @@
 import { Component, OnInit } from '@angular/core';
 import { MatDialog } from '@angular/material/dialog';
 import { DailogPopupComponent } from '../../shared/dialog-popup/dailog-popup.component';
+// import { projectDetailsData } from '../../project-details.component.spec.data';
```
Comment (Vinod-V3) [inline projects/project-player-library/src/lib/pages/details-page/details-page.component.ts:4]: remove commented code if its not needed

Comment (farhanp1502) [reply projects/project-player-library/src/lib/pages/details-page/details-page.component.ts:4]: done

File: projects/project-player-library/src/lib/shared/subtask-card/subtask-card.component.ts [hunk 11]
```diff
@@ -0,0 +1,88 @@
...
+  selector: 'lib-subtask-card',
+  templateUrl: './subtask-card.component.html',
+  styleUrl: './subtask-card.component.css'
+})
+export class SubtaskCardComponent {
+@Input() subTask: any;
+@Output() deleteSubTaskEvent = new EventEmitter<string>();
+@Output()  updateSubTaskStatusEvent = new EventEmitter<any>();
+
+subTaskOptions = [
```
Comment (Vinod-V3) [inline projects/project-player-library/src/lib/shared/subtask-card/subtask-card.component.ts:16]: Make these options come from config file

File: projects/project-player-library/src/lib/pages/task-details-page/task-details-page.component.ts [hunk 12]
```diff
@@ -1,20 +1,113 @@
...
-export class TaskdetailspageComponent implements OnInit{
-  constructor(private route:ActivatedRoute){}
+export class TaskDetailsPageComponent implements OnInit {
+  constructor(private route: ActivatedRoute, private dialog: MatDialog) {}
+  taskId: any;
+  selectedValue!: string;
+  textFormControl = new FormControl('');
+  task: any = {};
+  taskOptions = [
+    {
```
Comment (Vinod-V3) [inline projects/project-player-library/src/lib/pages/task-details-page/task-details-page.component.ts:20]: Options should come from config file

File: projects/project-player-library/src/lib/pages/main-player/main-player.component.css [hunk 13]
```diff
@@ -1,7 +1,7 @@
 .inner-outlet {
     max-width: 700px;
-    min-height: 100vh;
-    padding: 10px ;
+    min-height: 97.5vh;
```
Comment (Vinod-V3) [inline projects/project-player-library/src/lib/pages/main-player/main-player.component.css:3]: No need to add this change, revert this file change

File: projects/project-player-library/src/lib/pages/details-page/details-page.component.ts [hunk 14]
```diff
@@ -92,6 +94,7 @@ export class DetailsPageComponent implements OnInit {
       case "download":
         this.projectDetails.downloaded = true
         this.setActionsList()
+        this.toasterservice.showToast("success",2000,"top","right")
```
Comment (Vinod-V3) [inline projects/project-player-library/src/lib/pages/details-page/details-page.component.ts:97]: Follow camel case convention

File: projects/project-player-library/src/lib/shared/subtask-card/subtask-card.component.ts (hunk 11 above)
Comment (farhanp1502) [reply projects/project-player-library/src/lib/shared/subtask-card/subtask-card.component.ts:16]: done

File: projects/project-player-library/src/lib/pages/task-details-page/task-details-page.component.ts (hunk 12 above)
Comment (farhanp1502) [reply projects/project-player-library/src/lib/pages/task-details-page/task-details-page.component.ts:20]: done

File: projects/project-player-library/src/lib/pages/main-player/main-player.component.css (hunk 13 above)
Comment (farhanp1502) [reply projects/project-player-library/src/lib/pages/main-player/main-player.component.css:3]: done

File: projects/project-player-library/src/lib/pages/details-page/details-page.component.ts (hunk 14 above)
Comment (farhanp1502) [reply projects/project-player-library/src/lib/pages/details-page/details-page.component.ts:97]: done

File: projects/project-player-library/src/lib/pages/details-page/details-page.component.ts [hunk 15]
```diff
@@ -13,19 +14,20 @@ import { actions } from '../../constants/actionConstants';
 export class DetailsPageComponent implements OnInit {
   completedCount: number = 0;
   progressValue: number = 0;
+  showAllTasks: boolean = false;
   actionsList=[
```
[the remark this answers is not in the document]
Comment (farhanp1502) [reply projects/project-player-library/src/lib/pages/details-page/details-page.component.ts:18]: done

File: projects/project-player-library/src/lib/constants/actionConstants.ts [hunk 16]
```diff
@@ -36,5 +36,50 @@ export const actions = {
         icon: "check_circle",
         action: "downloaded",
         color: "green"
-    }
+    },
```
[the remark this answers is not in the document]
Comment (farhanp1502) [reply projects/project-player-library/src/lib/constants/actionConstants.ts:39]: done
